static-instrumenter/agent-instrumenter/src/main/java/io/opentelemetry/contrib/statical/instrumenter/Main.java

```diff
@@ -0,0 +1,157 @@
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.jar.JarFile;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Main {
+
+  private static final Logger logger = LoggerFactory.getLogger(Main.class);
+
+  private static final Main INSTANCE = new Main(new ClassArchive.Factory() {});
+
+  private final ClassArchive.Factory classArchiveFactory;
+
+  private final Map<String, byte[]> instrumentedClasses = new ConcurrentHashMap<>();
+  private final Map<String, byte[]> additionalClasses = new ConcurrentHashMap<>();
+
+  @SuppressWarnings("ThreadLocalUsage")
+  private final ThreadLocal<TransformedClass> currentClass = new ThreadLocal<>();
+
+  public static void main(String[] args) throws Exception {
+
+    if (args.length == 0) {
+      printUsage();
+      return;
+    }
+
+    File outDir = new File(args[0]);
+    if (!outDir.exists()) {
+      outDir.mkdir();
+    }
+
+    String classPath = System.getProperty("java.class.path");
+    logger.debug("Classpath (jars list): " + classPath);
+    String[] jarsList = classPath.split(File.pathSeparator);
+
+    getInstance().saveTransformedJarsTo(jarsList, outDir);
+  }
+
+  @SuppressWarnings("SystemOut")
+  private static void printUsage() {
+    System.out.println(
+        "OpenTelemetry Java Static Instrumenter\n"
+            + "Usage:\n"
+            + Main.class.getCanonicalName()
+            + " <output directory> (where instrumented archives will be stored)");
+  }
+
+  public static Main getInstance() {
+    return INSTANCE;
+  }
+
+  public static ClassFileTransformer getPreTransformer() {
+    return new PreTransformer();
+  }
+
+  public static ClassFileTransformer getPostTransformer() {
+    return new PostTransformer();
+  }
+
+  // for testing purposes
+  Main(ClassArchive.Factory classArchiveFactory) {
+    this.classArchiveFactory = classArchiveFactory;
+  }
+
+  // FIXME: java 9 / jmod support, proper handling of directories
+  // FIXME: jmod in particular introduces weirdness with adding helpers to the dependencies
+
+  /**
+   * Copies all class archives (JARs, WARs) to outDir. Classes that were instrumented and stored in
+   * instrumentedClasses will get replaced with the new version. All classes added to
+   * additionalClasses will be added to the new archive.
+   *
+   * @param outDir directory where jars will be written
+   * @throws IOException in case of file operation problem
+   */
+  public void saveTransformedJarsTo(String[] jarsList, File outDir) throws IOException {
+
+    for (String pathItem : jarsList) {
+      logger.info("Classpath item processed: " + pathItem);
+      if (shouldProcessItem(pathItem)) {
+        saveJarTo(new File(pathItem), outDir);
+      }
+    }
+  }
+
+  private static boolean shouldProcessItem(String pathItem) {
+    return (pathItem.endsWith(".jar") || pathItem.endsWith(".war"));
```

Comment:
We haven't taken this format into consideration during plugin development. I think it's best to focus on (fat) jars and war (since all our tests so far were performed for these formats) and leave ears for the future